src/System.IO.Pipelines/src/System/IO/Pipelines/Pipe.cs

```diff
@@ -60,15 +60,20 @@ public sealed partial class Pipe
         private PipeCompletion _writerCompletion;
         private PipeCompletion _readerCompletion;
 
-        // The read head which is the extent of the IPipelineReader's consumed bytes
+        // Stores the last examined position, used to calculate how much bytes were to release
+        // for back pressure management
+        private BufferSegment _lastExamined;
```

Comment:
How much do we care about increasing the size of the pipe with these fields?

Reply:
I care but I haven't found a more succinct way to implement this yet.